std/file.d

```diff
@@ -456,7 +456,7 @@ Params:
 Throws: $(D FileException) on error.
  */
 void append(R)(R name, const void[] buffer)
-    if (isInputRange!R && isSomeChar!(ElementEncodingType!R) || isSomeString!R)
+    if (_isNonAliasedCharInputRange!R)
```

Comment:
What we can find a good name for `StringTypeOfX(R) if (!isSomeString!R)` then we can add `!is(StringTypeOfX!R)` and `is(StringTypeOfX!R)`.